runbot_gitlab_minimal/models/runbot_repo.py

```diff
@@ -117,3 +120,89 @@ def github(self, cr, uid, ids, url, payload=None, ignore_errors=False,
+                super(RunbotBuild, self).github_status(cr, uid, ids,
+                                                       context=context)
+            if not build.repo_id.token:
+                continue
+            session = _get_session(build.repo_id.token)
+            try:
+                if is_merge_request:
+                    url = _get_url('/projects/:owner/:repo/merge_requests/'
+                                   '?iid=%s' % build.branch_id.branch_name,
+                                   build.repo_id.base)
+                    response = session.get(url)
+                    response.raise_for_status()
+                    json = response.json()[0]
+                    source_project_id = json['source_project_id']
+                    if source_project_id:
+                        url = _get_url('/projects/%s' % source_project_id,
+                                       build.repo_id.base)
+                        response = session.get(url)
+                        response.raise_for_status()
+                        json = response.json()
+                        base_url = (json['web_url'].replace('http://', '').
+                                    replace('https://', ''))
+                        _url = _get_url('/projects/:owner/:repo/statuses/%s' %
+                                        build.name, base_url)
+                desc = "runbot build %s" % (build.dest,)
+                if build.state == 'testing':
+                    state = 'running'
+                elif build.state in ('running', 'done'):
+                    state = 'failed'
+                    if build.result == 'ok':
+                        state = 'success'
+                    if build.result == 'ko':
+                        state = 'failed'
+                    if build.result == 'skipped':
+                        state = 'canceled'
+                    if build.result == 'killed':
+                        state = 'canceled'
+                    desc += " (runtime %ss)" % (build.job_time,)
+                else:
+                    continue
+                status = {
+                    "state": state,
+                    "target_url": "http://%s/runbot/build/%s" % (runbot_domain,
+                                                                 build.id),
+                    "description": desc,
+                    "context": "ci/runbot"
+                }
+                _logger.debug("gitlab updating status %s to %s", build.name,
+                              state)
+                response = session.post(_url, status)
+                response.raise_for_status()
+            except Exception:
+                _logger.exception('gitlab API error %s', url)
+
+
+class RunbotBranch(models.Model):
+    _inherit = "runbot.branch"
+
+    branch_url = fields.Char(compute='_get_branch_url')
+
+    @api.multi
+    def _get_branch_url(self):
+        _branch_urls = super(RunbotBranch, self)._get_branch_url(None, None)
```

Comment:
The first parameter is `field_name` you could use `branch_url` instead of `None`